src/SSW.Rewards.Application/Quizzes/Commands/SubmitUserQuiz/SubmitUserQuizCommand.cs

```diff
@@ -71,13 +75,22 @@ private void AddQuizAchievement(int achievementId, int userId)
             _context.UserAchievements.Add(quizCompletedAchievement);
         }
 
-        private void AddCompletedQuiz(int quizId, int userId)
+        private async void AddCompletedQuiz(int quizId, int userId, List<QuizAnswerDto> answers, bool passed)
         {
             CompletedQuiz c = new CompletedQuiz
             {
                 QuizId = quizId,
-                UserId = userId
+                UserId = userId,
+                Passed = passed
             };
+
+            foreach (var answer in answers)
+            {
+                var dbAnser = await _context.QuizAnswers.FirstOrDefaultAsync(q => q.Id == answer.SelectedAnswerId);
```

Comment:
change to .FirstAsync because it should never be null (and we would expect it to throw an error if it couldn't find it)

Reply:
Done